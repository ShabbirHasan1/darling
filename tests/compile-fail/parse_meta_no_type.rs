extern crate proc_macro;
use darling::{parse_meta, FromMeta};
use proc_macro::TokenStream;
use syn::ItemFn;

#[derive(Debug, FromMeta)]
struct MacroArgs {
    #[darling(default)]
    timeout_ms: Option<u16>,
    path: String,
}

pub fn your_attr(args: TokenStream, input: TokenStream) -> TokenStream {
    let args = parse_meta!(args);
    let _input = syn::parse_macro_input!(input as ItemFn);

    println!("{:?}", args.timeout_ms);
    unimplemented!()
}

fn main() {}
