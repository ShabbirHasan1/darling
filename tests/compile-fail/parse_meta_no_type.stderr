error[E0282]: type annotations needed
  --> tests/compile-fail/parse_meta_no_type.rs:14:9
   |
14 |     let args = parse_meta!(args);
   |         ^^^^
...
17 |     println!("{:?}", args.timeout_ms);
   |                      ---- type must be known at this point
   |
help: consider giving `args` an explicit type
   |
14 |     let args: _ = parse_meta!(args);
   |             +++
